nixos/modules/services/development/jupyter/default.nix

```diff
@@ -37,14 +45,36 @@ in
       '';
     };
 
-    # NOTE: We don't use top-level jupyter because we don't
-    # want to pass in JUPYTER_PATH but use .environment instead,
-    # saving a rebuild.
-    package = lib.mkPackageOption pkgs [ "python3" "pkgs" "notebook" ] { };
+    package = lib.mkPackageOption pkgs [
+      "python3"
+      "pkgs"
+      "jupyter"
+    ] { };
+
+    extraPackages = lib.mkOption {
+      type = lib.types.listOf lib.types.package;
+      default = [ ];
+      example = ''
+        [
+          pkgs.python3.pkgs.nbconvert
+          pkgs.python3.pkgs.playwright
+        ]
+      '';
+      description = ''Extra packages to be available in the jupyter runtime enviroment'';
+    };
+    extraEnvironmentVariables = lib.mkOption {
+      type = lib.types.attrsOf lib.types.anything;
```

Comment:
I think we should [inherit](https://github.com/NixOS/nixpkgs/blob/6c90912761c43e22b6fb000025ab96dd31c971ff/nixos/modules/config/shells-environment.nix#L41-L43) the `type` and `apply` from `environment.variables` like how [`environment.sessionVariables` does](https://github.com/NixOS/nixpkgs/blob/6c90912761c43e22b6fb000025ab96dd31c971ff/nixos/modules/config/system-environment.nix#L32).